Author: Luca Boccassi <luca.boccassi@gmail.com>
Description: Fix get_user_pages API usage for kernel >= 4.9
 From kernel 4.9 and newer (commit 768ae309a9) get_user_pages has changed.
 Patch NV_GET_USER_PAGES macro in common/inc/nv-mm.h to fix build failures.

--- a/nv-linux.h
+++ b/nv-linux.h
@@ -1828,7 +1828,13 @@ static inline NvU64 nv_node_end_pfn(int
 #endif /* defined(NV_GET_NUM_PHYSPAGES_PRESENT) */
 
 #if defined(NV_GET_USER_PAGES_REMOTE_PRESENT)
+#include <linux/version.h>
+#if LINUX_VERSION_CODE < KERNEL_VERSION(4,9,0)
     #define NV_GET_USER_PAGES           get_user_pages
+#else
+    #define NV_GET_USER_PAGES(start, nr_pages, write, force, pages, vmas) \
+        get_user_pages(start, nr_pages, write ? FOLL_WRITE : 0, pages, vmas)
+#endif
     #define NV_GET_USER_PAGES_REMOTE    get_user_pages_remote
 #else
     #define NV_GET_USER_PAGES(start, nr_pages, write, force, pages, vmas) \
